<h1> Lab5 </h1> 
Akshita Gundavarapu

In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import statsmodels.formula.api as smf

<h2> 1 Diamonds data </h2> 
<h3> 1.1 </h3> 

In [5]:
diamond = pd.read_csv("data /diamonds.csv.bz2", sep = "\t")
diamond.head(3)

carat      cut color clarity  depth  table  price     x     y     z
0   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31

In [6]:
diamond.shape

(53940, 10)

We have 53940 diamonds. 

In [8]:
diamond.describe()

carat         depth         table         price             x  \
count  53940.000000  53940.000000  53940.000000  53940.000000  53940.000000   
mean       0.797940     61.749405     57.457184   3932.799722      5.731157   
std        0.474011      1.432621      2.234491   3989.439738      1.121761   
min        0.200000     43.000000     43.000000    326.000000      0.000000   
25%        0.400000     61.000000     56.000000    950.000000      4.710000   
50%        0.700000     61.800000     57.000000   2401.000000      5.700000   
75%        1.040000     62.500000     59.000000   5324.250000      6.540000   
max        5.010000     79.000000     95.000000  18823.000000     10.740000   

                  y             z  
count  53940.000000  53940.000000  
mean       5.734526      3.538734  
std        1.142135      0.705699  
min        0.000000      0.000000  
25%        4.720000      2.910000  
50%        5.710000      3.530000  
75%        6.540000      4.040000  
max       58.900000     31.800000

<h3> 1.2 </h3>

In [9]:
diamond.cut.describe()

count     53940
unique        5
top       Ideal
freq      21551
Name: cut, dtype: object

In [10]:
diamond.cut.value_counts()

Ideal        21551
Premium      13791
Very Good    12082
Good          4906
Fair          1610
Name: cut, dtype: int64

<h2> Regression analysis </h2> 
<h3> 2.1 </h3> 

In [11]:
m = smf.ols("price ~ carat + cut", data = diamond).fit()
m.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  price   R-squared:                       0.856
Model:                            OLS   Adj. R-squared:                  0.856
Method:                 Least Squares   F-statistic:                 6.437e+04
Date:                Thu, 17 Feb 2022   Prob (F-statistic):               0.00
Time:                        01:07:26   Log-Likelihood:            -4.7142e+05
No. Observations:               53940   AIC:                         9.429e+05
Df Residuals:                   53934   BIC:                         9.429e+05
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
====================================================================================
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept        -3875.4697     40.408    -95.908      0.000   -3954.670   -3796.269
cut[T.Good]       1120.3319     43.499     25.755      0.000    1035.073    1205.591
cut[T.Ideal]      1800.9240     39.344     45.773      0.000    1723.809    1878.039
cut[T.Premium]    1439.0771     39.865     36.098      0.000    1360.941    1517.214
cut[T.Very Good]  1510.1354     40.240     37.528      0.000    1431.265    1589.006
carat             7871.0821     13.980    563.040      0.000    7843.682    7898.482
==============================================================================
Omnibus:                    14616.138   Durbin-Watson:                   1.027
Prob(Omnibus):                  0.000   Jarque-Bera (JB):           150962.278
Skew:                           1.007   Prob(JB):                         0.00
Kurtosis:                      10.944   Cond. No.                         18.7
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

<h3> 1.2 </h3> 
The reference category for cut is fair 

<h3> 1.3 </h3> 

Interpret the following coefficients:

(a) What is “carat” (correct value should be 7871)?

For every 1 usit increase in Carat there is a 7871 units in price given that the Cut remains the same.

(b) What is “cut[T:Ideal]” (correct value should be 1800)?

1800 is then the average difference in Price between the category for which the cut is ideal and all other cuts, given that Carat remains the same.

(c) How much more expensive are ideal-cut diamonds compared to very good-cut diamonds (in average) given they weight the same?

290 units more expensive


(d) What does your model predict–what is be the (average) price for 1ct premium cut diamond?
7871+ 1439 - 3875 = 5435 units in price